helpers/corpus.py

```diff
@@ -0,0 +1,89 @@
+import json
+import random
+from shared import config
+
+
+class Corpus:
+    def __init__(self):
+        self.chain = {}
+        self.order = 2
+        self.openingwords = {}
+
+        self.load_json(config['Qud install folder'] +
+                       "/CoQ_Data/StreamingAssets/Base/LibraryCorpus.json")
+
+    def generate_sentence(self, seed="") -> str:
+        if len(seed) == 0 or seed.isspace():
+            seed = self.openingwords[random.randint(
+                0, len(self.openingwords))]
+        list = []
+
+        for i in range(0, self.order):
+            list.append(seed.split(' ')[i])
+        for i in range(0, 100):
+            text = list[i]
+            for j in range(1, self.order):
+                text = f"{text} {list[i+j]}"
+                text2 = self.chain[text][random.randint(
+                    0, len(self.chain[text])-1)]
+                if text2 == "#MAKESECRET#":
```

Comment:
Since this is an internal signal, this could use a comment saying what it is used for